extension-manifest-v3/src/background/exceptions.js

```diff
@@ -30,15 +30,23 @@ function negateFilter(filter) {
   }
 }
 
+let exceptions = new Map();
+
+chrome.storage.local.get(['exceptions']).then((result) => {
+  exceptions = new Map(Object.entries(result.exceptions));
+});
+
 chrome.storage.onChanged.addListener(async (changes) => {
   if (!changes['exceptions']) {
     return;
   }
 
+  exceptions = new Map(Object.entries(changes['exceptions'].newValue || {}));
```

Comment:
This is overkill, as the exceptions is a plain object with keys, no need to transfer it to the Map instance.